### PR TITLE
Add Indeed OAuth Provider

### PR DESCRIPTION
**Pull request by sahapranta**
Hi,

I am requesting to add indeed OAuth Provider, Here is the doc

https://docs.indeed.com/dev/docs/log-in-with-indeed

https://docs.indeed.com/dev/docs/upgrade-oauth

### CODE REVIEW

File: src/Indeed/Provider.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+class Provider extends AbstractProvider
+{
+    public const IDENTIFIER = 'INDEED';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopeSeparator = ' ';
+
+    protected $scopes = ['email', 'offline_access'];
```
Comment (atymic) [inline src/Indeed/Provider.php:18]: Do we need `offline_access` token by default? I assume we can omit this in the provider

**Review by atymic — CHANGES_REQUESTED** [review]
See comments

File: src/Indeed/Provider.php (hunk 1 above)
Comment (sahapranta) [reply src/Indeed/Provider.php:18]: Hi @atymic, I don't know why but they require to have `offline_access` included when requesting for `email`

[https://docs.indeed.com/dev/docs/authentication-overview#scopes](https://docs.indeed.com/dev/docs/authentication-overview#scopes:~:text=Incremental%20authorization%20enables%20apps%20to%20request%20only%20the%20scopes%20that%20they%20need%2C%20and%20requires%20the%20offline_access%2C%20which%20maintains%20the%20permissions%20that%20your%20app%20has.%20For%20example%2C%20if%20your%20app%20requests%20the%20email%20scope%2C%20your%20app%20must%20also%20request%20the%20offline_access%20scope.)

Here is a screenshot from the above url.

image

File: src/Indeed/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+```php
+protected $listen = [
+    \SocialiteProviders\Manager\SocialiteWasCalled::class => [
+        // ... other providers
+        \SocialiteProviders\Indeed\IndeedExtendSocialite::class.'@handle',
+    ],
+];
+```
+
+### Usage
```
[the remark this answers is not in the document]
Comment (sahapranta) [reply src/Indeed/README.md:36]: Thanks for the mention & Updated the docs.

File: src/Indeed/Provider.php (hunk 1 above)
Comment (atymic) [reply src/Indeed/Provider.php:18]: Ahh ok.

**Comment by atymic** [comment]
Please PR to website repo

**Pull request merged**